src/soft-tabs-transform.coffee

```diff
@@ -0,0 +1,38 @@
+require('es6-shim')
+
+Point = require './point'
+Region = require './region'
+
+module.exports =
+class SoftTabsTransform
+  constructor: (@tabLength) ->
+    @softTab = " ".repeat(@tabLength)
+
+  initialize: (@source) ->
+
+  getRegions: ({sourceStartPosition}) ->
+    sourceEndPosition = @source.getEndPosition()
+    sourcePosition = sourceStartPosition
+    regions = []
```

Comment:
> I'm sort of having regrets now about suggesting we return more than one region at a time here. Sorry for the trouble, but I take it back. 

---------

> Sorry for the thrash on this, but I'm still getting my head wrapped around the problem.

No problem at all :smile: That was something I should have noticed myself, therefore no need to excuse :+1: As you point out, these are still early days and it's totally acceptable to overlook something and to make mistakes. Hopefully, they should be helpful to understand the problem better :wink: 

> Instead what we should do is define a startsWith method on layers that takes a string and an optional position. In this case we would simply ask if the underlying layer starts with a soft tab at our current position and if it does, consume it. Otherwise we'd return a region extending to the end of the line as you do. This ensures that we don't scan ahead any more than needed in the input, which will keep the streaming window as small as possible as we're constructing regions at each layer.

That sounds reasonable. The `startsWith` method could be a simple helper over `positionOf`, which already provides most of what we need. I'll revert the `getRegions` change altogether now, as I think we're better off addressing it when we have a real use-case (e.g. soft wrapping, as you suggest).